File: Go/StackArray.go
package main
import "fmt"

var top int = -1
var stack = [10]int{}

func push(int data){
	if top == 10 {
		fmt.Println("Stack Overflow")
	}
	top++
	stack[top] = data
}

func peek() int {
	if top == 10 {
		fmt.Println("Stack Overflow")
	}

	if top == -1 {
		fmt.Println("Stack Empty")
	}

	return stack[top]
}

func pop() int {
	if top == -1 {
		fmt.Println("Stack Empty")
	}
	data := stack[top]
	top--
	return data
}

func isEmpty() bool {
	if len(stack) == 0 {
		return true
	} else {
		return false
	}
}

func size() int {
	return len(stack)
}

func printList(){
	for i := 0; i < len(stack); i++ {
		fmt.Println(stack[i])
	}
}

func main() {
	fmt.Println(isEmpty())

	push(1)
	push(3)
	push(5)
	push(7)
	push(9)

	fmt.Println(isEmpty())

	fmt.Println(peek())
	fmt.Println(size())
	fmt.Println(pop())
	printList()
}

File: C#/StackArray.cs
using System;

public class StackArray
{
	int [] stack;
	int top = -1;

	public int peek() {
		if (top == -1) {
			Console.WriteLine("Stack Empty");
		}
		return stack[top];
	}

	public int pop() {
		if (top == -1) {
			Console.WriteLine("Stack Empty");
		}
		int index = top;
		top--;
		return stack[index];
	}

	public void push(int data) { 
		top++;
		stack[top] = data;
	}

	public bool isEmpty() {
		if (top == -1) { return true; }
		return false;
	}

	public int size() {
		return stack.Length;
	}

	public void printList() { 
		for (int i = 0; i < stack.Length;  i++) {
			Console.WriteLine(stack[i]);
		}
	}

	public static void Main(string[] args) {
		StackArray stack = new StackArray();

		Console.WriteLine(stack.isEmpty());
		stack.push(1);
        stack.push(3);
        stack.push(5);
        stack.push(7);
        stack.push(9);
        Console.WriteLine(stack.isEmpty());
        stack.printList();
		stack.pop();
		stack.printList();
    }
}
